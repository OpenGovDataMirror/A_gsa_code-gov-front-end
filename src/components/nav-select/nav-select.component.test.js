import React from 'react'
import { shallow } from 'enzyme'

import NavSelect from 'components/nav-select/nav-select.component'

const props = {
  handleChange: jest.fn(),
  pathname: '/page-2',
  pages: [
    { route: '/page-1', display: 'page 1' },
    { route: '/page-2', display: 'page 2' }
  ]
}

let wrapper
let instance
describe('components - NavSelect', () => {
  beforeEach(() => {
    wrapper = shallow(<NavSelect {...props} />)
    instance = wrapper.instance()
  })

  describe('handleChange', () => {
    it('should call `handleChange` with the value', () => {
      instance.handleChange({ target: { value: 'test-value' } })
      expect(props.handleChange).toBeCalledWith('test-value')
    })

    it('should not throw if `handleChange` not provided', () => {
      wrapper.setProps({ handleChange: undefined })
      expect(() => instance.handleChange({ target: { value: 'test-value' } })).not.toThrow()
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should get the current page based off the `pathname`', () => {
      expect(wrapper.find('select').prop('value')).toEqual('/page-2')
    })

    it('should map the `pages` as options', () => {
      expect(wrapper.find('option').length).toBe(props.pages.length)
      expect(
        wrapper
          .find('option')
          .at(0)
          .text()
      ).toBe(props.pages[0].display)
    })
  })
})
