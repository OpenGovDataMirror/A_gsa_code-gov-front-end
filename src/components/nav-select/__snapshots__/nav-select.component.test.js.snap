// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - NavSelect render should render correctly 1`] = `
<form
  className="usa-form maxw-none"
>
  <label
    className="usa-label usa-sr-only"
    htmlFor="options"
  >
    Sections Menu
  </label>
  <select
    className="usa-select font-body-3xs radius-md"
    id="options"
    onChange={[Function]}
    value="/page-2"
  >
    <option
      key="/page-1"
      value="/page-1"
    >
      page 1
    </option>
    <option
      key="/page-2"
      value="/page-2"
    >
      page 2
    </option>
  </select>
</form>
`;
